src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_consent_manager_impl.h

```diff
@@ -149,21 +148,21 @@ class RCConsentManagerImpl : public RCConsentManager {
    * acquiring resource
    * @param module_type Module type with consents
    */
-  Json::Value& GetModuleTypeConsentsOrAppend(
-      const std::string& policy_app_id,
-      const std::string& mac_address,
-      const std::string& module_type) const;
+  Json::Value& GetModuleTypeConsentsOrAppend(const std::string& policy_app_id,
+                                             const std::string& mac_address,
+                                             const std::string& module_type);
 
  private:
   application_manager::ApplicationManager& app_manager_;
-  resumption::LastState& last_state_;
+  resumption::LastStateWrapperPtr last_state_;
   const uint32_t period_of_consent_expired_;
   mutable sync_primitives::Lock dictionary_control_lock_;
   mutable sync_primitives::Lock remote_control_lock_;
   mutable sync_primitives::Lock device_applications_lock_;
   mutable sync_primitives::Lock applications_lock_;
   mutable sync_primitives::Lock app_consents_lock_;
   mutable sync_primitives::Lock module_consents_lock_;
+  Json::Value last_state_dictionary_;
```

Comment:
@YarikMamykin I think you should remove this variable as you can always get dictionary using accessor

Reply:
@AKalinich-Luxoft actually not always. Since 2dd6f73cf75d7c17d86e0313dde501d3766e61c2 `Json::Value& LastState::get_dictionary()` method is replaced with `Json::Value LastState::dictionary()` which returns copy of last state dictionary ( I guess it was done for thread safe access to dictionary ).
Since this change we have to remember changes made in RCConsentManager and update LastState dictionary when it is needed.

Reply:
@AByzhynar please check https://github.com/smartdevicelink/sdl_core/pull/2626/commits/c5c0f062625e5fb1485ebb97f4f3258aedcc8f2f